projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php

```diff
@@ -721,46 +790,55 @@ function wpcom_global_styles_is_previewing_premium_theme_without_premium_plan( $
 /**
  * Checks whether the site has access to Global Styles with a Personal plan as part of an A/B test.
  *
- * @todo: Update the code to support Atomic sites for A/B tests.
- *
- * @param  int $blog_id Blog ID.
  * @return bool Whether the site has access to Global Styles with a Personal plan.
  */
-function is_global_styles_on_personal_plan( $blog_id = 0 ) {
-	if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+function is_global_styles_on_personal_plan() {
+	// Environment guard: only WP.com or Atomic.
+	$is_wpcom  = defined( 'IS_WPCOM' ) && IS_WPCOM;
+	$is_atomic = defined( 'IS_ATOMIC' ) && IS_ATOMIC;
+	if ( ! $is_wpcom && ! $is_atomic ) {
 		return false;
 	}
 
-	if ( ! function_exists( '\ExPlat\assign_given_user' ) ) {
+	$connection_manager  = new Automattic\Jetpack\Connection\Manager();
+	$wpcom_blog_owner_id = $connection_manager->get_connection_owner_id();
+	$wpcom_blog_id       = $connection_manager->get_site_id();
+
+	if ( ! $wpcom_blog_owner_id || ! $wpcom_blog_id ) {
```

Comment:
There isn't a Jetpack connection on Simple sites, so I think we need a different logic for them.

Reply:
I'll create a logical branch then, on Simple sites I'll default to using the previous approach.